firestore/lib/src/model/filter.dart

```diff
@@ -0,0 +1,19 @@
+typedef FilterChangedCallback<T> = void Function(T? newValue);
+
+class Filter {
+  final String? city;
+  final int? price;
+  final String? category;
+  final String? sort;
+
+  bool get isDefault {
+    return (city == null && price == null && category == null && sort == null);
+  }
+
+  Filter({
+    this.city,
+    this.price,
+    this.category,
+    this.sort,
+  });
+}
```

Comment:
I always find myself doing this, too. Might be nice to have a docstring on the `Filter` class explaining why it exists.